pkg/app/api/analysisresultstore/filestore.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analysisresultstore
```

Comment:
How about `analysismetadatastore` because it is dealing with `AnalysisMetadata` model?

Reply:
I was thinking that this package will be able to handle the actual metrics data points in the future. But it can be separated, let me rename it

Reply:
I think this can be used to handle that case in the future too so it is fine to use the same package.
And I was thinking that we would use the same `AnalysisMetadata` model as well.

Inside that model, we have some fields like:

```
DataSourceType: EXTERNAL | IN_FILE
DataPoints: []
```

And the user can decide whether to store the metrics data to our file store or not, by configuring a field in their ANALYSIS stage.

Reply:
In that case, I would prefer that the name is `AnalysisResult`, not `AnalysisMetadata`. What about you?

Reply:
That looks good. Totally agree about that name.

Reply:
Like:

```go
type AnalysisResult struct {
	StartTime      time.Time
	DataSourceType string
	// Data points for each analysis
	MetricsResults map[string]DataPoint
}
```

Reply:
Alright!

Reply:
Looks nice. 
nit: `MetricsResults` -> `Metrics`